storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -870,17 +870,21 @@ bool xbcloud_put(Object_store *store, const std::string &container,
   xtrabackup_tablespaces.00000000000000000000) is uploaded to cloud storage to
   determine successful xbcloud "put" operation. */
   std::string last_file_prefix = backup_name + "/xtrabackup_tablespaces";
+  auto last_file_size = last_file_prefix.size();
   bool file_found = false;
-
-  if (!object_list.empty()) {
-    auto cur_file = object_list.back();
-    auto last_file_size = last_file_prefix.size();
-    if (cur_file.size() >= last_file_size &&
-        cur_file.substr(0, last_file_size).compare(last_file_prefix) == 0)
+  for (auto cur_file = object_list.end() - 1; cur_file >= object_list.begin();
```

Comment:
There are built-in ways on the std library to traverse a vector backward. Can you check https://www.cplusplus.com/reference/vector/vector/rbegin/

Reply:
thank you @altmannmarcelo for suggestion. using rbegin now.